schema/body_schema.go

```diff
@@ -50,6 +51,24 @@ func NewBodySchema() *BodySchema {
 	}
 }
 
+func (as *BodySchema) SortAttributes() []string {
```

Comment:
```suggestion
func (as *BodySchema) AttributeNames() []string {
```
I guess `SortedAttributeNames` would also be a reasonable name, but either way I think that the name should communicate that we're just _extracting_ attribute names, not modifying any kind of internal order.

Reply:
Apparently great minds think alike. You've already implemented these [here](https://github.com/hashicorp/hcl-lang/blob/4bf451bad31e52c5575261a27966b692f1c564de/decoder/body_candidates.go#L85-L105). Do you have a preference for that style vs this style? If we decide on an implementation, would it make sense to move the reference in `body_candidates` to use the interface method?

Reply:
Ah, right, I completely forgot about these functions! I don't have too strong opinion, but my temptation would be to use the functions inside the `decoder` package.

The assumption/expectation I'm working with is that (from consumer perspective, such as `terraform-schema`) when declaring the schema it shouldn't matter how attributes are sorted, it's only when you decode them when it matters.

I also went as far as "redeclaring" the `ExpressionConstraint` schema type to reflect the same train of thoughts there: https://github.com/hashicorp/hcl-lang/blob/4bf451bad31e52c5575261a27966b692f1c564de/decoder/expression_constraints.go#L9 but I think that for sorting alone we shouldn't need it.

Reply:
I've decided to leave them where they are for now to not complicate things further for this PR